plugins/nic/src/plugin.cpp

```diff
@@ -28,15 +28,33 @@ namespace {
 struct loader_args {
   located<std::string> iface;
   std::optional<located<uint32_t>> snaplen;
+  std::optional<location> emit_file_headers;
 
   template <class Inspector>
   friend auto inspect(Inspector& f, loader_args& x) -> bool {
     return f.object(x)
       .pretty_name("loader_args")
-      .fields(f.field("iface", x.iface), f.field("snaplen", x.snaplen));
+      .fields(f.field("iface", x.iface), f.field("snaplen", x.snaplen),
+              f.field("emit_file_headers", x.emit_file_headers));
   }
 };
 
+auto make_file_header(int snaplen, int linktype) -> pcap::file_header {
+  return {
+#ifdef PCAP_TSTAMP_PRECISION_NANO
```

Comment:
As discussed, please confirm that this works as expected.

Reply:
You found a bug.

Fixed in 5475a4ed32.